go/vt/vtgate/evalengine/translate_card.go

```diff
@@ -134,6 +134,8 @@ func (ast *astCompiler) cardExpr(expr Expr) error {
 		return ast.cardUnary(expr.Inner)
 	case *BitwiseNotExpr:
 		return ast.cardUnary(expr.Inner)
+	case *NotExpr:
+		return ast.cardUnary(expr.Inner)
```

Comment:
@systay I've included the fix for `NOT` here and also fixed up the types and updated it all with additional tests. It also adds the other logical operations like `AND` to the compiler with tests which we were missing as well.